compiler/rustc_driver_impl/src/args.rs

```diff
@@ -153,3 +153,40 @@ impl fmt::Display for Error {
 }
 
 impl error::Error for Error {}
+
+/// check if suspect is a confusables of any option or arg in rustc or fake_args
+pub(crate) fn check_confusables(suspect: &str, fake_args: &[&str]) -> Option<String> {
```

Comment:
Question: is there a reason why this is in `args.rs`? Or can this just be inlined as idk a nested helper inside of `warn_on_confusing_output_filename_flag` below?